salt/grains/core.py

```diff
@@ -1109,7 +1109,9 @@ def _virtual(osdata):
         if ("virtual_subtype" not in grains) or (grains["virtual_subtype"] != "LXC"):
             if os.path.isfile("/proc/1/environ"):
                 try:
-                    with salt.utils.files.fopen("/proc/1/environ", "r") as fhr:
+                    with salt.utils.files.fopen(
+                        "/proc/1/environ", "r", errors="ignore"
+                    ) as fhr:
```

Comment:
I'm thinking this could lead to some data loss.  What if we just used "surrogateescape"?  This way, we can write the exact same contents to another location if need be using the same `errors=`
```suggestion
                    with salt.utils.files.fopen(
                        "/proc/1/environ", "r", errors="surrogateescape"
                    ) as fhr:
```

Reply:
It makes no any sense to get `surrogateescapes`, just check how the data from the file is used after

Reply:
@MKLeb I've put `surrogateescape` to the other part of this change related to `kernelparams` as this is the only place reflecting to the values shown in grains. For the rest part of this change it makes no any sense to replace `ignore` with `surrogateescape`.